src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.cs

```diff
@@ -134,12 +134,31 @@ protected void SkipToAndParseCode(Func<HtmlSymbol, bool> condition)
                 {
                     if (last != null)
                     {
+                        if (startOfLine && last.Type == HtmlSymbolType.WhiteSpace)
```

Comment:
Any chance there's more than one symbol waiting to be output? Concern is using the wrong generator for non-whitespace.